### PR TITLE
Update build process for OpenBSD

### PR DESCRIPTION
**Pull request by tuftedocelot**
OpenBSD builds appear to have gone a bit stale so make the changes to allow building.

Currently:
* i686 builds and runs
* x86_64 yields kernel panics
* GCC toolchain builds
* Clang toolchain build fails

Redo of #11378 to satisfy linting and remove an unnecessary OpenBSD-specific define check with LocalSocket

### CODE REVIEW

File: Userland/Libraries/LibCore/System.cpp [hunk 1]
```diff
@@ -118,6 +118,8 @@ ErrorOr<void> sigaction(int signal, struct sigaction const* action, struct sigac
 
 #ifdef __APPLE__
 ErrorOr<sig_t> signal(int signal, sig_t handler)
+#elif defined(__OpenBSD__)
```
Comment (bgianfo) [inline Userland/Libraries/LibCore/System.cpp:121]: We should avoid re-declaring the same signature as above.. how about instead we just modify the above `#ifdef` to something like: 

```cpp
#if defined(__APPLE__) || defined(__OpenBSD__)
```

Same goes for the header as well?

File: Userland/Libraries/LibCore/LocalSocket.cpp [hunk 2]
```diff
@@ -14,6 +14,8 @@
 
 #if defined(__FreeBSD__)
 #    include <sys/ucred.h>
+#elif defined(__OpenBSD__)
+#    include <sys/socket.h>
```
Comment (bgianfo) [inline Userland/Libraries/LibCore/LocalSocket.cpp:18]: We already `#include <sys/socket.h>` above, line 13, why is this duplicate include also needed?

**Review by bgianfo — CHANGES_REQUESTED** [review]
Looks good! Just two small issues.

File: Userland/Libraries/LibCore/LocalSocket.cpp (hunk 2 above)
Comment (tuftedocelot) [reply Userland/Libraries/LibCore/LocalSocket.cpp]: It isn't. I missed removing that when I was mucking about with the creds/finding the pid

**Review by bgianfo — APPROVED** [review]

**Comment by tuftedocelot** [comment]
With https://github.com/SerenityOS/serenity/pull/11398/commits/a8f52576b299c9db32aeeeecc5cbca9802203e3f, I think that'll conclude this initial PR to improve builds on OpenBSD. If I make any further changes re the list above, I'll open separate, more specific PR

**Comment by tuftedocelot** [comment]
@bgianfo, whenever you have a second, can you take a quick look at https://github.com/SerenityOS/serenity/pull/11398/commits/5f6cb1d9284fff1aa7dfc1df5ef62e65bf8c768a? I realized that the underscore for the gz paths was a bad idea because they are the full path, not just the file. Hopefully my final change isn't too far into bikeshed territory...

**Review by linusg — CHANGES_REQUESTED** [review]
Please squash fixup commits or remove no longer relevant commits. Currently the initial commit adds something that is later removed.

**Comment by tuftedocelot** [comment]
indeed, thanks @linusg!

**Comment by trflynn89** [comment]
FYI, on Ubuntu, I'm now getting this error:
```
gzip: invalid option -- 'o'
```

For reference:
```
$ gzip --version
gzip 1.10
Copyright (C) 2018 Free Software Foundation, Inc.
Copyright (C) 1993 Jean-loup Gailly.
This is free software.  You may redistribute copies of it under the terms of
the GNU General Public License <https://www.gnu.org/licenses/gpl.html>.
There is NO WARRANTY, to the extent permitted by law.

Written by Jean-loup Gailly.
```

Could we use this [CMake function](https://cmake.org/cmake/help/latest/command/file.html#archiving)? It might require a bump to `cmake_minimum_required` though.

**Comment by BertalanD** [comment]
@trflynn89 There's an issue about it, turns out that has its own pitfalls too. See #9866

**Comment by tuftedocelot** [comment]
Looks like `gzip -d -c file.gz > FILENAME` ought to be the most portable option here. I'll get a new PR going

**Pull request merged**